### PR TITLE
A mixed bag of minor fixes, mostly related to language use (I mean English, not C++)

### PR DESCRIPTION
**Pull request by reunanen**
Not sure all of these are for the better, but think that on average they should be.

### CODE REVIEW

File: CppCoreGuidelines.md [hunk 1]
```diff
@@ -8379,7 +8379,7 @@ The problem is that the unqualified and qualified lookups both have uses.
 ##### Reason
 
 The `&` operator is fundamental in C++.
-Many parts of the C++ semantics assumes its default meaning.
+Many parts of the C++ semantics assume its default meaning.
```
Comment (jwakely) [inline CppCoreGuidelines.md:8382]: I personally like this, but opinions differ on whether "semantics" should take a singular or plural verb.

File: CppCoreGuidelines.md [hunk 2]
```diff
@@ -16654,7 +16656,7 @@ but that should be done only when the called function is supposed to modify the
 
 ##### Enforcement
 
-* Flag function that does not modify an object passed by  pointer or reference to non-`const`
+* Flag a function that does not modify an object passed by  pointer or reference to non-`const`
```
Comment (jwakely) [inline CppCoreGuidelines.md:16659]: Harmless, but there's an unnecessary extra space before "pointer".

File: CppCoreGuidelines.md [hunk 3]
```diff
@@ -17262,7 +17264,7 @@ This `Addable` violates the mathematical rule that addition is supposed to be co
 
 ##### Note
 
-The ability to specify a meaningful semantics is a defining characteristic of a true concept, as opposed to a syntactic constraint.
+The ability to specify meaningful semantics is a defining characteristic of a true concept, as opposed to a syntactic constraint.
```
Comment (jwakely) [inline CppCoreGuidelines.md:17267]: See previous comment about semantics.

File: CppCoreGuidelines.md [hunk 4]
```diff
@@ -20612,7 +20614,7 @@ and errors (when we didn't deal correctly with semi-constructed objects consiste
             : mx(check_size(x))
             , my(check_size(y))
             // now we know x and y have a valid size
-            , data(mx * my * sizeof(int)) // will throw std::bad_alloc on error
+            , data(mx * my) // will throw std::bad_alloc on error
```
Comment (jwakely) [inline CppCoreGuidelines.md:20617]: This example no longer matches the "bad" one before it. I don't think removing the `sizeof(int)` from only one of them is a good change. It will make people wonder if the reason this one is good is because it uses `int` not `char` (but that doesn't seem relevant to this rule).

**Review by jwakely — COMMENTED** [review]
Most of this looks good to me, I've added some comments in places.

File: CppCoreGuidelines.md (hunk 1 above)
Comment (reunanen) [reply CppCoreGuidelines.md:8382]: I don't disagree – just thought that it might read better this way. I can of course revert this change.

File: CppCoreGuidelines.md (hunk 2 above)
Comment (reunanen) [reply CppCoreGuidelines.md:16659]: Fixed.

File: CppCoreGuidelines.md (hunk 4 above)
Comment (reunanen) [reply CppCoreGuidelines.md:20617]: > This example no longer matches the "bad" one before it.

Yes, true. That was indeed my intention (to make this one "better" than the "bad" one, also in this respect – because apparently the picture has pixels whose type is `int`).

I have now changed also the "bad" example to use `int*`.

But I can of course just as well revert this whole change, if that sounds better?

Comment (reunanen) [reply CppCoreGuidelines.md:20617]: At least `char` sounds wrong to me (as per [SL.str.5](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rstr-byte)). If not `int`, shouldn't the type be `byte`?

Shouldn't at least the "good" examples be "good" from all possible points of view, and not just the rule being discussed?

File: CppCoreGuidelines.md (hunk 1 above)
Comment (reunanen) [reply CppCoreGuidelines.md:8382]: (Actually, to me it sounds like "parts" is key here, and not "semantics." But I'm not a native speaker, and I [have had trouble with similar matters also before](https://english.stackexchange.com/questions/5378/a-number-of-questions-has-been-or-have-been-asked). :)

File: CppCoreGuidelines.md (hunk 4 above)
Comment (jwakely) [reply CppCoreGuidelines.md:20617]: I don't know what the "design" of the example is, so I don't know why it uses units of `sizeof(int)` but allocated as `char`. I agree it seems questionable. Maybe `uint8_t` would be better (because you can't do arithmetic on `byte` and many image processing transformations involve arithmetic).

File: CppCoreGuidelines.md (hunk 1 above)
Comment (jwakely) [reply CppCoreGuidelines.md:8382]: Oh, you're absolutely right! It should definitely be a plural verb here. I got distracted by "semantics" but that's not actually relevant here.

File: CppCoreGuidelines.md (hunk 4 above)
Comment (reunanen) [reply CppCoreGuidelines.md:20617]: But it really seems that the pixel values are supposed to be `int`. So I think doing arithmetic on `uint8_t` looks very wrong, too.

File: CppCoreGuidelines.md (hunk 3 above)
Comment (reunanen) [reply CppCoreGuidelines.md:17267]: Ok, so this one is actually different [from the other one](https://github.com/isocpp/CppCoreGuidelines/pull/1776#discussion_r615805908).

I'm fine with reverting this change.

Comment (reunanen) [reply CppCoreGuidelines.md:17267]: So... Should I just revert this change?

**Comment by hsutter** [comment]
Editors call: Thanks!
